common/djangoapps/pipeline_mako/templates/static_content.html

```diff
@@ -70,21 +70,16 @@
 <%def name="require_module(module_name, class_name)">
     <script type="text/javascript">
         (function (require) {
-            % if settings.REQUIRE_DEBUG:
-                require(['${module_name}'], function (${class_name}) {
-                    ${caller.body()}
-                });
-            % else:
-                ## The "raw" parameter is specified to avoid the URL from being further maninpulated by
-                ## static_replace calls (as woudl happen if require_module is used within courseware).
-                ## Without specifying "raw", a call to static_replace would result in the MD5 hash being
-                ## being appended more than once, causing the import to fail in production environments.
-                require(['${staticfiles_storage.url(module_name + ".js") + "?raw"}'], function () {
-                    require(['${module_name}'], function (${class_name}) {
-                        ${caller.body()}
-                    });
-                });
-            % endif
+            require(['${staticfiles_storage.url(module_name + ".js") + "?raw"}'], function (${class_name}) {
```

Comment:
This isn't what I had in mind. The optimized factory file needs to only be loaded in production mode (i.e. not debug mode), and it needs to be loaded via a script tag and not a ``require`` call. The former will ensure that the file is loaded synchronously, while the latter will load it asynchronously. I would probably put this script tag as the first line inside the ``require_module`` definition.

With that change in place, you can then leave this script tag pretty much as is. You can just delete the ``else`` clause entirely, and then keep the ``if`` clause but render it unconditionally. In debug mode this will just require the module name as normal. In production mode, it will also require the module name, but it will already have been loaded by the bundled production file.

Let's chat in person tomorrow if this isn't clear.